### PR TITLE
URLPopover: use new placement prop instead of legacy position prop

### PR DESCRIPTION
**Pull request by ciampo**
## What?

Part of #44401

Use the new `placement` prop instead of the legacy `position` prop to define the `Popover`'s placement in the `URLPopover` component.

## Why?

With the recent refactor of the `Popover` component to `floating-ui`, we're in the process of refactoring all of its usages to the new `placement` prop (native to `floating-ui`). See #44401 for more details

## How?

- Introduced new `placement` prop for the `URLPopover` component
- marked the `position` prop as deprecated — the prop, however, is still supported by the component
- internally, the `URLPopover` component uses the new `placement` prop for the `Popover` component, also thanks to the `positionToPlacement` prop that is being exported from the `Popover` component

See this [conversion table](https://github.com/WordPress/gutenberg/blob/0a9402ac623876cc2a29d0f4a72eaefba3310501/packages/components/src/popover/utils.ts#L17-L71) that we're currently using to map `position` values to `placement` values.

## Testing Instructions

- Add an image block, click on the "Insert from URL" button. Make sure that the URL popover shows up in the same position as on `trunk`
- Add a Social Link block, then add a generic social link. Click on the link icon button in the editor, and make sure that the URL Popover shows up in the same position as on trunk

## Screenshots or screencast 

https://user-images.githubusercontent.com/1083581/198674577-d29ae53a-e8b0-45a9-b739-c50f4e1e94bc.mp4

## ✍️ Dev Note 

With the recent refactor of the `Popover` component to use the `floating-ui` library, the `placement` prop has become the recommended way to determine how the component positions with respect to its anchor. While the older `position` prop is still supported, we're making changes towards removing all of its usages in the Gutenberg repository and deprecating it.

As part of these efforts, the `position` prop has been marked as deprecated on the following components:
- `Dropdown` from the `@wordpress/components` package
- `URLPopover` component from the `@wordpress/block-editor` package

Consumers of these components should be using the new `placement` prop instead.

### CODE REVIEW

File: packages/block-editor/src/components/url-popover/index.js
Comment (tyxla) [inline packages/block-editor/src/components/url-popover/index.js]: Isn't this a breaking change? Should we still pass a deprecated `position` and pass it as a `popover` prop in case someone uses this component outside of the repo, but still relies on the legacy prop?

In any case, this might be worth a CHANGELOG entry.

Comment (ciampo) [reply packages/block-editor/src/components/url-popover/index.js]: Good point, I guess that approach was indeed introducing a breaking change!

I've pushed some changes and adopted a more conservative approach — see the updated PR description for more details

File: packages/block-editor/src/components/url-popover/index.js
Comment (tyxla) [inline packages/block-editor/src/components/url-popover/index.js:49]: Extreme nit, but with that order of logic, if someone passes both `position` and `placement`, the deprecated prop will take precedence. Should we do this only if `placement` is not specified?

**Review by tyxla — APPROVED** [review]
LGTM 👍 

Just left a minor nitty question, other than that this is good to go 🚀

File: packages/block-editor/src/components/url-popover/index.js
Comment (ciampo) [reply packages/block-editor/src/components/url-popover/index.js:49]: Since I moved the default value from `position` to `placement`, the `placement` prop can't really be `undefined` in the function's body. Therefore, in order to maintain backwards compat, I decided to implement the current logic.

I could refactor the logic and:

- move the default value assignment of `placement` from the prop destructuring to later in the function body
- make it so that when both `placement` and `position` are defined, `placement` takes the precedence

Let me know if this sounds like a good idea to you! If so, I'll go ahead and implement it

Comment (tyxla) [reply packages/block-editor/src/components/url-popover/index.js:49]: What you suggested sounds good! I'll leave it to your judgment whether to implement it before merging.

Comment (ciampo) [reply packages/block-editor/src/components/url-popover/index.js:49]: Pushed https://github.com/WordPress/gutenberg/pull/44391/commits/e149eefca998202e325c5c824b388ab2982885aa with those changes, let me know if it looks good to you!

**Review by tyxla — APPROVED** [review]
Thanks for the extra comments. Looks great 🚀

**Pull request merged**
